merlin/study/study.py

```diff
@@ -413,23 +415,21 @@ def expanded_spec(self):
             )
 
         # write expanded spec for provenance
-        with open(expanded_filepath, "w") as f:  # pylint: disable=C0103
+        with open(self.special_vars["MERLIN_SPEC_ARCHIVED_COPY"], "w") as f:  # pylint: disable=C0103
             f.write(result.dump())
 
         # write original spec for provenance
-        result = MerlinSpec.load_spec_from_string(result.dump())
-        result.path = expanded_filepath
-        name = result.description["name"].replace(" ", "_")
-        self.write_original_spec(name)
+        self.write_original_spec()
 
         # write partially-expanded spec for provenance
         partial_spec = deepcopy(self.original_spec)
+        result = MerlinSpec.load_spec_from_string(result.dump())
```

Comment:
what are these lines doing?

Reply:
I didn't write these lines myself, I just copied them from the existing code and moved them down to organize the code better. It looks like before this line, result is building the entire expanded spec and then creating a MerlinSpec object from all the info loaded in. I'm not sure why it's necessary to create an entirely new MerlinSpec object here if result is already a MerlinSpec object... I'll try removing this and see if it impacts the partial spec creation at all

Reply:
After commenting them out, the partial yaml file is still outputting the same information so I don't think these lines are necessary. I'm removing them

Reply:
oop I lied, setting the path of the result is necessary for testing so I'm putting that line back in there